src/nupic/regions/PyRegion.cpp

```diff
@@ -532,26 +544,47 @@ void PyRegion::write(capnp::AnyPointer::Builder& proto) const
 void PyRegion::read(capnp::AnyPointer::Reader& proto)
 {
 #if !CAPNP_LITE
+
+  class Helper
+  {
+  public:
+    static kj::Array<capnp::word> flatArrayFromReader(
+      PyRegionProto::Reader& reader)
+    {
+      // NOTE: this requires conversion to builder, which incurs an additional
+      // copy, because readers aren't supported by capnp::messageToFlatArray
+      capnp::MallocMessageBuilder builder;
+      builder.setRoot(reader); // copy
+      return capnp::messageToFlatArray(builder); // copy
+    }
+
+    static PyObject* pyBytesFromFlatArray(kj::Array<capnp::word> flatArray)
+    {
+      kj::ArrayPtr<kj::byte> byteArray = flatArray.asBytes();
+      // Copy from array to PyObject so that we can pass it to the Python layer
+      py::String pyRegionBytes((const char *)byteArray.begin(),
+                               byteArray.size()); // copy
+      return pyRegionBytes.release();
+    }
+  };
+
+  // Cast AnyPointer::Reader to PyRegionProto::Reader
   PyRegionProto::Reader pyRegionReader = proto.getAs<PyRegionProto>();
 
-  // See PyRegion::read implementation for reference
+  // TODO Need to look into reducing the number of copy operations since regions
```

Comment:
Not sure what to make of this. Should we see if there is a way to do this or do we know there isn't currently a way to do so?